src/LiveComponent/src/Hydration/DoctrineEntityHydrationExtension.php

```diff
@@ -81,14 +81,44 @@ public function dehydrate(object $object): mixed
 
     private function objectManagerFor(string $class): ?ObjectManager
     {
-        if (!class_exists($class)) {
+        if (class_exists($class)) {
+            // Keep the standard way for a class
+            // todo cache/warmup an array of classes that are "doctrine objects"
+            foreach ($this->managerRegistries as $registry) {
+                if ($om = $registry->getManagerForClass($class)) {
+                    return self::ensureManagedObject($om, $class);
+                }
+            }
+
             return null;
         }
 
-        // todo cache/warmup an array of classes that are "doctrine objects"
-        foreach ($this->managerRegistries as $registry) {
-            if ($om = $registry->getManagerForClass($class)) {
-                return self::ensureManagedObject($om, $class);
+        if (interface_exists($class)) {
+            // special handler for interfaces
+            // For use cases : @see https://symfony.com/doc/current/doctrine/resolve_target_entity.html
+            // As today, getManagerForClass don't resolve nicely aliased interfaces
+            // The workaround is to enum over each object manager and trying to get metadata
+            // The metadata are indeed, resolved nicely
+            // Fore more details :
+            // @see \Doctrine\ORM\Tools\ResolveTargetEntityListener
```

Comment:
Why don't we use the ResolveTargetentityListener  ?

Reply:
Because at that time, i had not digged enough about how it was working under the hood :)

So, you're right, i will refactore it ;

Also, discussing with @Kocal, there was also the possibility to write another specific hydrator ; but it will imply to duplicate the most technical code of the current Doctrine Hydrator so i'm more in favor of updating the current one.

Do you have an opinion about this ?

Reply:
I'm not sure we should have specific code when it's not our "core" feature in UX component, and there is a bit "too many of this" in LiveComponent for my taste 😅 

What I would love is _**if**_ we could leverage or reuse existing code either in Doctrine, in Serializer, ...

But If we cannot, yep i'd agree with you: updating the current one feels much better. 

Also I suggest we do not implement the cache at the same time because, as we talked once or twice before, as soon with hit 3.0 we will start working "from scrartch" on a brand new metadata/factory system, and the whole hydrator process here is probably the first part of LiveComponent that would use it :)